src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -624,23 +629,39 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
         sprintf(keyS, "OFI-%d", rank);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Put(MPIDI_Global.kvsname, keyS, val), pmi);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Commit(MPIDI_Global.kvsname), pmi);
-        MPIDI_OFI_PMI_CALL_POP(PMI_Barrier(), pmi);
+
+        for (i = 0; i < size; i++) {
+            if (MPIDI_CH4_rank_is_local(i, MPIR_Process.comm_world)) {
```

Comment:
Is the locality information available at this point? When I looked at this before, a major problem was that locality was set up much later.